tests/FsCheck.Test/Runner.fs

```diff
@@ -1,5 +1,150 @@
+        let cmpTestData = function 
+            { NumberOfTests = nt1; NumberOfShrinks = ns1; Stamps = sx1; Labels = lx1 }, 
+            { NumberOfTests = nt2; NumberOfShrinks = ns2; Stamps = sx2; Labels = lx2 } -> 
+                nt1 = nt2 && ns1 = ns2 && Enumerable.SequenceEqual (sx1, sx2) && Enumerable.SequenceEqual (lx1, lx2)
+        let cmpOutcome = function
+            | Outcome.Timeout _, Outcome.Timeout _ -> true
+            | Outcome.Exception ex1, Outcome.Exception ex2 -> ex1.Equals ex2
+            | Outcome.False, Outcome.False -> true
+            | Outcome.True, Outcome.True -> true
+            | Outcome.Rejected, Outcome.Rejected -> true
+            | _, _ -> false
+        member __.Result = result.Value
+        member __.Sink = sink
+        override __.Equals other =
+            match other with
+            | :? ProbeRunner as pr -> 
+                let resultEq = 
+                    match pr.Result, __.Result with
+                    | TestResult.True (td1, so1), TestResult.True (td2, so2) -> 
+                        so1 = so2 && cmpTestData (td1, td2)
+                    | TestResult.False (td1, oa1, sa1, o1, r1), TestResult.False (td2, oa2, sa2, o2, r2) -> 
+                        cmpTestData (td1, td2) && 
+                        cmpOutcome (o1, o2) && 
+                        Enumerable.SequenceEqual (oa1, oa2) && 
+                        Enumerable.SequenceEqual (sa1, sa2) &&
+                        r1 = r2
+                    | TestResult.Exhausted td1, TestResult.Exhausted td2 -> cmpTestData (td1, td2)
+                    | _, _-> false
+                resultEq && Enumerable.SequenceEqual (pr.Sink, sink)
+            | _ -> false
+        interface IRunner with
+            override __.OnStartFixture _ = ()
+            override __.OnArguments (ntest, args, _) = 
+                sink.Add ntest
+                sink.AddRange args
+            override __.OnShrink(args, _) = 
+                sink.Add -1
+                sink.AddRange args
+            override __.OnFinished(_, testResult) = 
+                result <- Some testResult
+
+
+    let check arb body config =
+        let p = Prop.forAll arb body
+        let seed = Random.create () |> Some
+        printfn "seed %A" seed
+        let runner1 = ProbeRunner () 
+        FsCheck.Runner.check { config with Replay = seed; Runner = runner1; ParallelRunConfig = Some { MaxDegreeOfParallelism = Environment.ProcessorCount } } p
+        let runner2 = ProbeRunner () 
+        FsCheck.Runner.check { config with Replay = seed; Runner = runner2; ParallelRunConfig = None } p
+        runner1.Equals runner2
+            
+    [<Fact>]
+    let ``parallelTest produces same sequence as test on success`` () =
+        let body i = true
+        let arb = Arb.from<int>
+        let config = { Config.Quick with MaxTest = 30000; EndSize = 30000 }
+        if not <| check arb body config then failwith "can't find the way to call xunit assertion lol"
```

Comment:
Most of the tests use Unquote, `open Swensen.Unquote` then you can use `test <@ boolean assertion @>`. But I'm actually tiring of all these dependencies for very little benefit, esp. since Unquote doesn't have .NET standard build yet for example, so I'm happy with just throwing an exception like you did.